californium-core/src/main/java/org/eclipse/californium/core/CoapServer.java

```diff
@@ -135,72 +135,76 @@ public CoapServer(int... ports) {
 	 * {@link NetworkConfig#getStandard()} is used.
 	 * @param ports the ports to bind to
 	 */
-	public CoapServer(NetworkConfig config, int... ports) {
-		
+	public CoapServer(final NetworkConfig config, final int... ports) {
+
 		// global configuration that is passed down (can be observed for changes)
 		if (config != null) {
 			this.config = config;
 		} else {
 			this.config = NetworkConfig.getStandard();
 		}
-		
+
 		// resources
 		this.root = createRoot();
 		this.deliverer = new ServerMessageDeliverer(root);
-		
-		CoapResource well_known = new CoapResource(".well-known");
-		well_known.setVisible(false);
-		well_known.add(new DiscoveryResource(root));
-		root.add(well_known);
-		
+
+		CoapResource wellKnown = new CoapResource(".well-known");
+		wellKnown.setVisible(false);
+		wellKnown.add(new DiscoveryResource(root));
+		root.add(wellKnown);
+
 		// endpoints
-		this.endpoints = new ArrayList<Endpoint>();
+		this.endpoints = new ArrayList<>();
 		// sets the central thread pool for the protocol stage over all endpoints
 		this.executor = Executors.newScheduledThreadPool(//
 				config.getInt(NetworkConfig.Keys.PROTOCOL_STAGE_THREAD_COUNT), //
 				new Utils.NamedThreadFactory("CoapServer#")); //$NON-NLS-1$
 		// create endpoint for each port
-		for (int port:ports)
+		for (int port : ports) {
 			addEndpoint(new CoapEndpoint(port, this.config));
+		}
 	}
-	
-	public void setExecutor(ScheduledExecutorService executor) {
-		
-		if (this.executor!=null) this.executor.shutdown();
-		
+
+	public void setExecutor(final ScheduledExecutorService executor) {
+
+		if (this.executor != null) {
+			this.executor.shutdown();
```

Comment:
Why is the executor shutdown()? 
Is the executor intended to be "exclusive" for californium?
My understanding was, that I can use it to manage the threading of the callback and my own application logic.
If you shutdown(), you may shutdown more the expected.